index.js

```diff
@@ -141,6 +145,39 @@ function wrap (argv, env, workingDir) {
   return unwrap
 }
 
+// handle special cases of shebang, e.g., the
+// 2>/dev/null; exec "`dirname "$0"`/node" "$0" "$@"
+// approach used in Node 5.8.0.
+function fixShebang (options, workingDir) {
+  if (isWindows) return
+  // currently, the only edge-case that we solve
+  // is for Node > v5.8.0.
+  if (/^v0\..*$/.test(process.version)) return
```

Comment:
This might be dangerous, because it mixes the installed version of Node.js with the one of the current process together. If there’s any chance that a user installs Node >= 5.8.0 and executes the shipped npm executable from a v0.x process, this will fail.